smithy-codegen-core/src/main/java/software/amazon/smithy/codegen/core/directed/GenerateResource.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.codegen.core.directed;
+
+import software.amazon.smithy.codegen.core.CodegenContext;
+import software.amazon.smithy.model.shapes.ResourceShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+
+/**
+ * Directive used to generate a resource.
+ *
+ * @param <C> CodegenContext type.
+ * @param <S> Codegen settings type.
+ * @see DirectedCodegen#generateResource
+ */
+public final class GenerateResource<C extends CodegenContext<S, ?>, S> extends ShapeDirective<ResourceShape, C, S> {
```

Comment:
nit: these all have action names, but they don't actually perform actions, they just hold arguments used to perform those actions

Reply:
They are tied 1:1 with the methods invoked on DirectedCodegen. I considered putting "Context" on the end or something, but with the generics, it makes the methods _really_ long.

Reply:
I went ahead and added a Directive suffix to these classes.